tests/rptest/wasm/topics_result_set.py

```diff
@@ -28,9 +28,15 @@ def __init__(self,
         self.value = value
         self.offset = offset
 
+    def hash_without_offset(self):
+        return hash((self.topic, self.partition, self.key, self.value))
+
     def __eq__(self, o):
-        return self.topic == o.topic and self.partition == o.partition \
-            and self.key == o.key and self.value == o.value
+        return hash(self) == hash(o)
```

Comment:
This will work fine 99.99% of the time for the purposes of test code, but technically you could have a hash collision (same for where hash_without_offset is used as a dict key later).

The stricter way to do it would be to have a `to_tuple` (and maybe a `to_tuple_without_offset` method), and then your equality becomes `self.to_tuple() == o.to_tuple()` and your __hash__ becomes `hash(self.to_tuple())`.

I don't mind if you want to merge this in current form though, in the context of the test it's highly unlikely to cause an issue.

Reply:
Yup i agree, will make this change then omit the final change in the PR to be ready for merge